### DISCUSSION TITLE
Port for twitter - current world situation

### DISCUSSION THREAD

**Discussion by BeatSkip** [ThioJoe/YT-Spammer-Purge#712] [Ideas]
Hey, I would like to propose a port of the core functionality to twitter, as I think this would have a DEFINITE good impact in the current world situation. 
(let's just keep the dickwad i'm talking about aside for now as it's all clear and we all know who, what, how and why. And focus on the technical aspects of the proposal).  I would definitely be able to contribute, but probably not realistic on my own.

But as a basis, you'd give a certain hashtag and it'll go over X amount of recent tweets and the magic picks the spam and spammers and auto-report etc etc.

Anyone up for it?

Edit: as for the reason why it would have a good impact, personally I can't see *any* valid reasons for spamming on twitter for any reason whatsoever. As any spam or mass twittering for any reason goes fundamentally against the whole fundamental core idea of twitter. And especially in situations like now, twitter is completely overrun with utter garbage and any possible good impact twitter could have on the world is completely and utterly blown out of the water with the utter shite spamfest going around right now. (again, yes, the topic is clear, but in general I can't see any possible good reason for spamming in any context 'ever')

**Comment by TechStudent10** [comment]
I haven't seen a scammer on Twitter, ever. Oh wait I don't use Twitter.

Anyways, I could agree with this proposal, but also not. Maybe it's because I haven't seen this first hand. Porting this to Twitter would be relatively easy, it's just a matter of porting over the logic, tearing out the Google API, bring in the Twitter API, refactor the code and you've got `Twitter-Spammer-Purge`.

**Comment by Rairye** [comment]
@TechStudent11 I think there have been reports of impersonation scams on Twitter. 
@BeatSkip What kind of Twitter spam are you talking about? Are you referring to the fact that some people use trending hashtags to try to get more interaction?

**Reply by BeatSkip** [reply]
Well, I'm not a regular twitter user at all. But I do like it when stuff is going down to see the latest updates from people on the ground. But just when you try to find anything outside your own 'twitter sphere' as it were. It's basically a wild west of obvious bots ranging from spotting the exact same wording over numerous accounts to just rampant spamming of anything and everything and all grades of bots in between. Stuff 'exactly' like the comments that this tool excels at picking out on YouTube. I think it sucks, because in between the spam, I do see a valuable advantage that twitter has over other social media. And there is of course the large scale propaganda going on from each and every corner of the world, but to me that's partially a matter of 'use your actual thinking brain and bullshit spidey sense on social media'. But to get to that point, you first have to scroll through mountains of just pure crap spam. (and no I don't have friends on twitter as I'm absolutely not interested in sharing every brainfart with my friends. And do not really care for every brainfart from my friends, or worse, vague people you know or people you know, know) so I'm not really pointing this idea towards that aspect of twitter)

**Reply by Rairye** [reply]
Filtering on Twitter is relatively easy and convenient if you are trying to keep spam (unwanted content) out of your feed.

**Comment by ikkysleepy** [comment]
The limiting factor here is the twitter API  limits.

**Comment by BeatSkip** [comment]
> The limiting factor here is the twitter API  limits.

Yeah, I was kind of thinking about that. But I could only find references to posting limits, nothing about reporting limits

**Reply by SeqBra** [reply]
You sure there's a reporting API then? I'd assume all of twitter's public APIs have public documentation

**Reply by TechStudent10** [reply]
Found this in the Twitter Developer Portal: https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/mute-block-report-users/overview

**Reply by TechStudent10** [reply]
However, this is the V1 API.
